src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,115 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        size = 0;
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size >= (table.length) * DEFAULT_LOAD_FACTOR) {
+            resize();
+        }
+        int index = getHashCode(key);
+        Node<K, V> newNode = new Node<>(key, value);
```

Comment:
Using the diamond operator without type parameters is not recommended as it can lead to unchecked conversion warnings. Use `new Node<K, V>(...)` instead of `new Node<>(...)` to maintain type safety.